includes/feature_heatmap_form.inc

```diff
@@ -181,12 +181,14 @@ function generate_heatmap_values() {
     $z = array();
     $text = array();
     $max_id_len = 0;
+    global $base_url;
     foreach ($expression_data as $feature_id => $values) {
       // obtain max feature id length
       if (strlen($feature_id) > $max_id_len) {
         $max_id_len = strlen($feature_id);
       }
-      $y[] = '<a href="/feature/' . $feature_id . '">' . $feature_id . '</a>';
+      $nid = chado_get_nid_from_id('feature', $feature_id);
```

Comment:
The issue is that 'feature_id' is NOT actually the feature ID. 
$feature_id is set with 
`$feature_ids = preg_split("/[\s,]+/", $heatmap_feature_uniquename, -1, PREG_SPLIT_NO_EMPTY);`
So it's basically the unique name of the feature. So 'chado_get_nid_from_id' fails.

Reply:
hmmm good catch. I think the tripal 3 version does a much better job at linking out. I'll take a look at it there and see if we can port back.